packages/framework/presence/src/presenceDatastoreManager.ts

```diff
@@ -105,7 +103,13 @@ export class PresenceDatastoreManagerImpl implements PresenceDatastoreManager {
 		private readonly clientSessionId: ClientSessionId,
 		private readonly runtime: IEphemeralRuntime,
 		private readonly presence: PresenceManagerInternal,
+		systemWorkspaceDatastore: SystemWorkspaceDatastore,
+		systemWorkspace: PresenceStatesEntry<PresenceStatesSchema>,
 	) {
+		// eslint-disable-next-line @typescript-eslint/consistent-type-assertions
+		this.datastore = { "system:presence": systemWorkspaceDatastore } as PresenceDatastore;
```

Comment:
How does the concept of datastore here equates to runtime concept of datastore? And If I understood you right yesterday - single (runtime) datastore gets partitioned/namespaced into multiple workspaces?

Reply:
There is no direct relation to container runtime datastores. Datastore is just the appropropriate generic term for both of them.

Correct, for a single runtime there is a single over-arching datastore that includes all of the workspace "sub"-datastores.
[Presence Design Document - Datastore Structure](https://microsoft.sharepoint.com/:w:/t/FluidFramework/EZalAnQjd7NKrwAyan3kmVUBN42t6j9l9bnHJgiQvXmpzw?e=dIwtWI&nav=eyJoIjoiODkxNzQ5MDUifQ)

Reply:
Ah Ok, thank you. My suggestion here is to find some other term - data segment, data partition, data module, data block, data shard, data node. Whatever else. It's not that straightforward to explain existing datastore, so adding two different meanings to it may cause further confusion.